Trustroots/trustroots
# Use local circle images

We've been using external CDN service for free for many years to host tribe images and easily generate new sizes on demand. They've changed their free plan to a much more limited version, as well our traffic is bigger now (we consume 4-5 GB / month on tribe images) so hosting it isn't possible anymore to use Uploadcare without paying.

Since in the future we want folks to create new circles manually, it's convenient to take a step towards having these files in a folder.

For now, we generate new image sizes manually with a script, but we could in the future use these same functions to generate sizes on image upload — or even on-demand via URLs.

#### Proposed Changes

* Add a script to generate circle image sizes from local files:
  - In jpg and webp formats
  - Few different sizes
* Use local files in UI (across React & Angular) 
* In Mongo DB model: remove `image_UUID:tring` field, and add `image:boolean` field

TODO:
- [x] Do changes to server's deploy script to pull in actual images (stored at https://github.com/Trustroots/circle-images)

#### Testing Instructions

* Add images to new public/uploads-circle folder and run the bin script - you should get new file sizes and formats generated
* Observe circles at the site: all images should work as before, no visible changes
